Readers/interface/setJetBtagScore.h

```diff
@@ -0,0 +1,34 @@
+#ifndef TallinnNtupleProducer_Readers_setJetBtagScore_h
+#define TallinnNtupleProducer_Readers_setJetBtagScore_h
+
+
+
+#include "TallinnNtupleProducer/Objects/interface/RecoJetAK4.h"
+#include "TallinnNtupleProducer/Objects/interface/RecoLepton.h"
+
+#include <assert.h>                                              // assert()
+#include <vector>                                                // std::vector
+
+template <typename T>
+void
+setJetBtagScore(std::vector<T> & particles, const std::vector<RecoJetAK4> & jets)
+{
+  Int_t njets = jets.size();
+
+  for ( T & particle : particles )
+  {
+    T * particlePtr = &particle;
+    Int_t jetIdx = particlePtr->jetIdx();
+    if ( jetIdx >=0 &&  jetIdx <njets)
+    {
+      double btag = jets.at(jetIdx).BtagCSV();
+      particlePtr->setjetBtagCSV(btag);
+    }
+    else
+    {
+      particlePtr->setjetBtagCSV(-1);
+    }
+  }
+}
```

Comment:
Proposing a few changes: avoid intermediate conversion to pointers; change the if-else block such that `RecoLepton::setjetBtagCSV()` is called only once; prefer to use `const` where possible; check for NaNs like before. So something like this:
```cpp
setJetBtagScore(std::vector<T> & particles,
                const std::vector<RecoJetAK4> & jets)
{
  const Int_t njets = jets.size();
  for ( T & particle : particles )
  {
    const Int_t jetIdx = particle.jetIdx();
    double btag = ( jetIdx >=0 &&  jetIdx <njets) ? jets.at(jetIdx).BtagCSV() : -1;
    if(std::isnan(btag))
    {
      btag = -2;
    }
    particle.setjetBtagCSV(btag);
  }
}
```

Reply:
done, it is also test with previous result and getting same result